dts/bindings/flash_controller/zephyr,disk-flash.yaml

```diff
@@ -0,0 +1,51 @@
+# SPDX-License-Identifier: Apache-2.0
+
+description: Emulates a flash using a disk
+
+compatible: "zephyr,disk-flash"
+
+include: base.yaml
+
+properties:
+  size:
+    type: int
+    description: flash capacity in bits
+    required: true
+  disk-name:
+    type: string
+    description: the name of the underlying disk to be used
+    required: true
+  disk-sector-size:
+    type: int
+    description: the sector size in byte of the underlying disk to be used
+    required: true
+  disk-offset:
+    type: int
+    description: >
+      offset on the disk in bytes. If the offset is 10 the 10th byte will
+      be the first one used.
+    default: 0
+  init-level:
+    type: string
+    description: >
+      The initialization level of this driver. It is important, that the underlying disk has
+      already been initialized.
+    default: "APPLICATION"
+  init-priority:
+    required: true
+    type: int
+    description: >
+      The initialization priority of this driver. It is important, that the underlying disk has
+      already been initialized.
+  write-block-size:
+    type: int
+    description: size of a flash write block in bytes
+    default: 1
+  erase-value:
```

Comment:
Same question- why would a user need to configure the erase value here?

Reply:
My thought was that a user might want a defined value in the memory when it is deleted. Would you propose to just set this to 0?